tests/components/tod/test_binary_sensor.py

```diff
@@ -22,885 +20,890 @@ def mock_legacy_time(legacy_patchable_time):
-        self.hass.config.longitude = 18.98583
-
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_setup(self):
-        """Test the setup."""
-        config = {
-            "binary_sensor": [
-                {
-                    "platform": "tod",
-                    "name": "Early Morning",
-                    "after": "sunrise",
-                    "after_offset": "-02:00",
-                    "before": "7:00",
-                    "before_offset": "1:00",
-                },
-                {
-                    "platform": "tod",
-                    "name": "Morning",
-                    "after": "sunrise",
-                    "before": "12:00",
-                },
-            ]
-        }
-        with assert_setup_component(2):
-            assert setup.setup_component(self.hass, "binary_sensor", config)
-
-    def test_setup_no_sensors(self):
-        """Test setup with no sensors."""
-        with assert_setup_component(0):
-            assert setup.setup_component(
-                self.hass, "binary_sensor", {"binary_sensor": {"platform": "tod"}}
-            )
-
-    def test_in_period_on_start(self):
-        """Test simple setting."""
-        test_time = datetime(2019, 1, 10, 18, 43, 0, tzinfo=self.hass.config.time_zone)
-        config = {
-            "binary_sensor": [
-                {
-                    "platform": "tod",
-                    "name": "Evening",
-                    "after": "18:00",
-                    "before": "22:00",
-                }
-            ]
-        }
-        with patch(
-            "homeassistant.components.tod.binary_sensor.dt_util.utcnow",
-            return_value=test_time,
-        ):
-            setup_component(self.hass, "binary_sensor", config)
-            self.hass.block_till_done()
-
-        state = self.hass.states.get("binary_sensor.evening")
+@pytest.fixture(autouse=True)
+def setup_fixture(hass):
+    """Set up things to be run when tests are started."""
+    hass.config.latitude = 50.27583
+    hass.config.longitude = 18.98583
+
+
+async def test_setup(hass):
+    """Test the setup."""
+    config = {
+        "binary_sensor": [
+            {
+                "platform": "tod",
+                "name": "Early Morning",
+                "after": "sunrise",
+                "after_offset": "-02:00",
+                "before": "7:00",
+                "before_offset": "1:00",
+            },
+            {
+                "platform": "tod",
+                "name": "Morning",
+                "after": "sunrise",
+                "before": "12:00",
+            },
+        ]
+    }
+    with assert_setup_component(2):
+        assert await async_setup_component(hass, "binary_sensor", config)
+
+
+async def test_setup_no_sensors(hass):
+    """Test setup with no sensors."""
+    with assert_setup_component(0):
+        assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)